From 9ef88aec4102e4cf935a300ee885e9026497e16d Mon Sep 17 00:00:00 2001
From: Torstein Krause Johansen <torstein@escenic.com>
Date: Tue, 24 May 2016 14:32:13 +0200
Subject: [PATCH] LUCENE-7298: './bin/solr start' now leaves a tidy command
 prompt

The newline after the user message "...Happy searching!" must be
echoed outside of the sub shell because of the spinner to have the
intended effect.
---
 solr/bin/solr | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/solr/bin/solr b/solr/bin/solr
index aaf2f3d..ee92d69 100755
--- a/solr/bin/solr
+++ b/solr/bin/solr
@@ -1444,11 +1444,13 @@ function launch_solr() {
           fi
         else
           SOLR_PID=`ps auxww | grep start\.jar | grep -w $SOLR_PORT | grep -v grep | awk '{print $2}' | sort -r`
-          echo -e "\nStarted Solr server on port $SOLR_PORT (pid=$SOLR_PID). Happy searching!\n"
+          echo -e "\nStarted Solr server on port $SOLR_PORT (pid=$SOLR_PID). Happy searching!"
           exit # subshell!
         fi
       done) &
       spinner $!
+      # need final newline to leave the cli prompt intact
+      echo ""
     else
       echo -e "NOTE: Please install lsof as this script needs it to determine if Solr is listening on port $SOLR_PORT."
       sleep 10
-- 
2.8.1

